backend/open_webui/utils/tools.py

```
@@ -2,111 +2,203 @@
+    Dict,
+    List,
+    Tuple,
+    Union,
+    Optional,
+    Type,
+)
 from functools import update_wrapper, partial
 
 
 from fastapi import Request
 from pydantic import BaseModel, Field, create_model
```

Ruff (0.8.2)

Redefinition of unused `BaseModel` from line 9

Remove definition: `BaseModel`

(F811)